src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/WorkbookRepository.kt

```diff
@@ -40,54 +41,53 @@ class WorkbookRepository(
             title = bookCollection.titleKey,
             sort = bookCollection.sort,
             chapters = constructBookChapters(bookCollection),
-            subtreeResources = resourceRepository.getSubtreeResourceInfo(bookCollection)
+            subtreeResources = db.getSubtreeResourceInfo(bookCollection)
         )
     }
 
     private fun constructBookChapters(bookCollection: Collection): Observable<Chapter> {
-        val collections = collectionRepository
-            .getChildren(bookCollection)
-            .flattenAsObservable { list -> list.sortedBy { it.sort } }
-
-        val chapters = collections
-            .concatMapEager { constructChapter(it).toObservable() }
-
-        return chapters.cache()
+        val coldObservable = Observable.fromCallable {
+            db.getChildren(bookCollection)
+                .flattenAsObservable { list -> list.sortedBy { it.sort } }
+                .concatMapEager { constructChapter(it).toObservable() }
+        }
+        return coldObservable
+            .flatMap { it }
+            .cache()
     }
 
     private fun constructChapter(chapterCollection: Collection): Single<Chapter> {
-        return contentRepository
-            .getCollectionMetaContent(chapterCollection)
+        return db.getCollectionMetaContent(chapterCollection)
             .map { metaContent ->
                 Chapter(
                     title = chapterCollection.titleKey,
                     sort = chapterCollection.sort,
                     resources = constructResourceGroups(chapterCollection),
                     audio = constructAssociatedAudio(metaContent),
                     chunks = constructChunks(chapterCollection),
-                    subtreeResources = resourceRepository.getSubtreeResourceInfo(chapterCollection)
+                    subtreeResources = db.getSubtreeResourceInfo(chapterCollection)
                 )
             }
     }
 
     private fun constructChunks(chapterCollection: Collection): Observable<Chunk> {
-        val contents = contentRepository
-            .getByCollection(chapterCollection)
-            .flattenAsObservable { list -> list.sortedBy { it.sort } }
-            .filter { it.labelKey != "chapter" } // TODO: filter by something better
-
-        val chunks = contents
-            .map {
-                Chunk(
-                    title = it.start.toString(),
-                    sort = it.sort,
-                    audio = constructAssociatedAudio(it),
-                    resources = constructResourceGroups(it),
-                    text = textItem(it)
-                )
-            }
-
-        return chunks.cache()
+        val coldObservable = Observable.fromCallable {
```

Comment:
same as above, how about "sortedChunks"?

Reply:
Switched to Observable.defer